mods/beds/functions.lua

```diff
@@ -7,17 +7,43 @@ if enable_respawn == nil then
 end
 
 -- Helper functions
+local facedir_types = {
+	colorwallmounted = 8, -- 3 bits are used for facedir
+	colorfacedir = 32     -- 5 bits are used for facedir
+}
 
 local function get_look_yaw(pos)
 	local n = minetest.get_node(pos)
-	if n.param2 == 1 then
-		return pi / 2, n.param2
-	elseif n.param2 == 3 then
-		return -pi / 2, n.param2
-	elseif n.param2 == 0 then
-		return pi, n.param2
+	local face_dir = n.param2
```

Comment:
Best to rename this variable to not confuse it with 'facedir', as later it may be set to param2 % 4 which is a simple rotation aorund the y axis. Maybe call it 'rotation'?